src/fpm/manifest/example.f90

```diff
@@ -61,6 +62,11 @@ subroutine new_example(self, table, error)
            call syntax_error(error, "Could not retrieve example name")
            return
         end if
+        if(.not.is_fortran_name(to_fortran_name(self%name)))then
+           call syntax_error(error, 'manifest file syntax error: example name must be composed only of &
+           &alphanumerics, "-" and "_"  and start with a letter::'//self%name)
+           return
+        endif
```

Comment:
The generation of this error message could be put into a separate subroutine. Also, we can add more meaningful context to the error message by marking the first offending character in the name as well. We could generate an error message like the one below here:

```
example name must be composed only of alphanumerics, "-" and "_"  and start with a letter
  _project
--^
```

like it is done for file parsing errors or errors in the version number:

https://github.com/fortran-lang/fpm/blob/5617e65751193afb3c88fbc6cb8a32822ff52ffa/src/fpm/error.f90#L109-L110